vex-riscv/src/ffi/impl.cpp

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: 2022 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "VVexRiscv.h"
+#include "verilated.h"
+#include "interface.h"
+
+extern "C" {
+  VVexRiscv* vexr_init();
+  void vexr_shutdown(VVexRiscv *top);
+  void vexr_step(VVexRiscv *top, const INPUT *input, OUTPUT *output);
+}
+
+
+VVexRiscv* vexr_init()
+{
+  return new VVexRiscv();
+}
+
+void vexr_shutdown(VVexRiscv *top)
+{
+  delete top;
+}
+
+void vexr_step(VVexRiscv *top, const INPUT *input, OUTPUT *output)
```

Comment:
I think you should tweak this to

```c
void vexr_step(VVexRiscv *top, const bool *advance_time, const INPUT *input, OUTPUT *output)
```

Then something like:

```c
  top->clk = true;
  top->eval();

  if (advance_time){
    // Produce falling edge (and therefore a rising edge next step). I.e., if
    // `advance_time` is false, inputs will get reevaluated. This can be useful
    // for combinatorial simulation.
    top->clk = false;
    top->eval();
  }
```

This way you can selectively update INPUT / OUTPUT. Maybe we should have a call on this btw :)

Reply:
Given that we've talked about this offline I think we can mark this resolved.